### PR TITLE
allow delegate to return a nil for `interstitialViewAtIndex`.

### PR DESCRIPTION
**Pull request by mr-fixit**
nothing to add here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,8 @@ The Example project uses [Carthage](https://github.com/Carthage/Carthage) to int
 
 Then, in your checkout of the `NYTPhotoViewer` repo, run `carthage checkout --use-submodules`.
 
+Then run `scripts/bootstrap`.
```
Comment (ZevEisenberg) [inline README.md:26]: To make it friendlier to beginners:

```suggestion
Then run `./scripts/bootstrap`.
```

Also, I _think_ the bootstrap script should obviate the need for the Carthage checkout from the previous line? Not sure, though.

File: NYTPhotoViewer/NYTPhotosViewController.m [hunk 2]
```diff
@@ -579,63 +579,36 @@ - (NSInteger)totalItemCount {
...
-    BOOL isPhotoAvailableAtIndex = true;
-    if ([self.dataSource respondsToSelector:@selector(isPhotoAtIndex:)]) {
-        isPhotoAvailableAtIndex = [self.dataSource isPhotoAtIndex:afterIndex];
-    }
-    if (isPhotoAvailableAtIndex) {
-        return [self newPhotoViewControllerForPhoto:[self.dataSource photoAtIndex:afterIndex] atIndex:afterIndex];
-    } else {
-        return [self newViewControllerAtIndex:afterIndex];
-    }
+    return [self nextViewControllerFromIndex:viewController.photoViewItemIndex delta: 1 stopBeforeIndex: self.totalItemCount];
```
Comment (ZevEisenberg) [inline NYTPhotoViewer/NYTPhotosViewController.m:611]: Obj-C style nit
```suggestion
    return [self nextViewControllerFromIndex:viewController.photoViewItemIndex delta:1 stopBeforeIndex:self.totalItemCount];
```

File: NYTPhotoViewer/NYTPhotosViewController.m [hunk 3]
```diff
@@ -579,63 +579,36 @@ - (NSInteger)totalItemCount {
...
-    if ([self.dataSource respondsToSelector:@selector(isPhotoAtIndex:)]) {
-        isPhotoAvailableAtIndex = [self.dataSource isPhotoAtIndex:beforeIndex];
-    }
-    if (isPhotoAvailableAtIndex) {
-        return [self newPhotoViewControllerForPhoto:[self.dataSource photoAtIndex:beforeIndex] atIndex:beforeIndex];
-    } else {
-        return [self newViewControllerAtIndex:beforeIndex];
-    }
+- (UIViewController *)pageViewController:(UIPageViewController *)pageViewController viewControllerBeforeViewController:(UIViewController <NYTPhotoViewerContainer> *)viewController {
+    return [self nextViewControllerFromIndex:viewController.photoViewItemIndex delta: -1 stopBeforeIndex: -1];
```
Comment (ZevEisenberg) [inline NYTPhotoViewer/NYTPhotosViewController.m:607]: Obj-C style nit
```suggestion
    return [self nextViewControllerFromIndex:viewController.photoViewItemIndex delta:-1 stopBeforeIndex:-1];
```

File: README.md (hunk 1 above)
Comment (mr-fixit) [reply README.md:26]: Good question. I tried it with a fresh `clone` and the answer is "bootstrap is sufficent".

File: NYTPhotoViewer/NYTPhotosViewController.h [hunk 4]
```diff
@@ -317,7 +317,7 @@ extern NSString * const NYTPhotosViewControllerDidDismissNotification;
  *
  *  @return A `UIView`.
  */
-- (UIView *)photosViewController:(NYTPhotosViewController *)photosViewController interstitialViewAtIndex:(NSUInteger)index;
+- (UIView * _Nullable)photosViewController:(NYTPhotosViewController *)photosViewController interstitialViewAtIndex:(NSUInteger)index;
```
[the remark this answers is not in the document]
Comment (mr-fixit) [reply NYTPhotoViewer/NYTPhotosViewController.h:320]: done

File: NYTPhotoViewer/NYTPhotosViewController.m (hunk 2 above)
Comment (mr-fixit) [reply NYTPhotoViewer/NYTPhotosViewController.m:611]: done (everywhere)

File: NYTPhotoViewer/NYTPhotosViewController.m (hunk 3 above)
Comment (mr-fixit) [reply NYTPhotoViewer/NYTPhotosViewController.m:607]: done (everywhere)

**Pull request merged**